packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -426,24 +432,93 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
         }
     }
 
+    private tooltip(x: number, y: number): void {
+        const xPos = this.positionXMove;
+        const timeForX = this.getTimeForX(xPos);
+        let timeLabel: string | undefined = timeForX.toString();
+        if (this.props.unitController.numberTranslator) {
+            timeLabel = this.props.unitController.numberTranslator(timeForX);
+        }
+        const chartWidth = this.lineChartRef.current.chartInstance.width;
+        const arraySize = this.state.xyData.labels.length;
+        const index = Math.max(Math.round((xPos / chartWidth) * (arraySize - 1)), 0);
+        const points: any = [];
+        let zeros = 0;
+        this.state.xyData.datasets.forEach((d: any) => {
+            // In case there are less data points than pixels in the chart,
+            // calculate nearest value.
+            const adjustedY = d.data[index];
+            const rounded = isNaN(adjustedY) ? 0 : (BIMath.round(Number(adjustedY) * 100)/BIMath.round(100));
+            if (this.state.xyData.datasets.length === 1 || rounded > 0) {
+                const point: any = {
+                    label: d.label,
+                    color: d.borderColor,
+                    background: d.backgroundColor,
+                    value: rounded.toString(),
+                };
+                points.push(point);
+            }
+            else {
+                zeros += 1;
```

Comment:
It gives odd results, like if there are only two values and they are both zero, it shows only '2 others: 0'. Or if there are two values and one is zero, '1 other: 0'.

Also it doesn't protect against large tooltip if there are many non-zero values.

I would suggest the following algorithm: Compile all the points and sort them. Then only if there are more points than a certain threshold (e.g. 10+1) *** , show only the top 10, and show '(num - 10) others'.

We may also keep special handling for zeros, e.g.: with 69 data points, 32 non-zero and 37 zero:

<top 10>
22 others: non-zero
37 others: 0

*** threshold is 10+1 because if there are exactly 11 points, it's better to show them all than to show top 10 and '1 other' since it takes the same space anyways...

Reply:
Sorry, I didn't notice that you can have a scrollbar to show all non-zero data points. So I think it would be good enough to just have a threshold for a number of series where, below than number, we always include all data points regardless of the yValue.

Reply:
Good suggestion. Done.